migrations/2023-08-24-074840_create_active_oracles_table/up.sql

```diff
@@ -1,5 +1,9 @@
 CREATE TABLE active_oracles (
-    address CHAR(42) PRIMARY KEY,
+    address BYTEA PRIMARY KEY,
     chain_id INTEGER NOT NULL,
-    specification JSONB NOT NULL
+    specification JSONB NOT NULL,
+    answer_tx_hash BYTEA,
+
+    CONSTRAINT address_length CHECK (LENGTH(address) = 20),
```

Comment:
Shouldn't it be 42 here as the max length?

Reply:
Well 42 bytes is what it takes to represent an address in hex string form, with the 0x prefix, which is the usual "human" representation. In reality, an address is 20 bytes long. The hex string length of 42 comes from the fact that in hex each byte must be represented by 2 hex digits (1 hex digit can represent 16 values max, 0 to 15, which translates to what 4 bits can represent, which is why 2 hex characters are required to represent a byte). That means that the 20 bytes are represented in string form by 40 characters. Then you add the 0x prefix (to make it clear what's following is an hex string) and you end up with 42 chars. Here we're heavily optimizing space as we're storing the bytes directly, without converting them from/to string